Source/Core/DolphinQt2/MainWindow.cpp

```diff
@@ -86,6 +91,35 @@
 #include "UICommon/X11Utils.h"
 #endif
 
+#if defined(__unix__) || defined(__unix) || defined(__APPLE__)
+void MainWindow::OnSignal()
+{
+  static bool once = false;
+
+  if (once)
+    return;
+
+  const auto message =
```

Comment:
This message should be printed in the signal handler instead to make sure it will always be shown even when the UI completely deadlocks.